src/index.d.ts

```diff
@@ -539,6 +555,57 @@ export function statusMatrix(args: {
   pattern: string;
 }): Promise<StatusMatrix>;
 
+export function tag(args: {
+  core?: string;
+  fs?: any;
+  dir: string;
+  gitdir?: string;
+  name: string;
+  value?: string;
+  force?: boolean;
+}): Promise<void>;
+
+export function annotatedTag(args: {
+  core?: string;
+  fs?: any;
+  dir: string;
+  gitdir?: string;
+  name: string;
+  object: {
+    message: string;
+    object?: string;
```

Comment:
Now I'm wondering about consistency of parameter names here. As for `annotatedTag` command, I thought the parameter to pass the value of tag should be named `object.object` like above, so that users can pass an object which represents tag object as is, but this name is inconsistent with the one in the original `tag` command, which is named `value`. I just used `value` because this accepts not only object ids but also refs, like `git tag` does.

However I think `annotatedTag` should still accept refs, for the use of just creating annotated tags. What can we do about this? Should we rename the original one to `object`, or leave it as it is now?

Reply:
Instead of having a bit object, I split it up into its individual parts (message, tagger, object, signingKey) to more closely match the design of `commit`. The way I see it creating an annotated tag is pretty similar to creating a commit. What do you think about that?

Reply:
> The way I see it creating an annotated tag is pretty similar to creating a commit.

I think that's a good design goal.